agent/job_runner.go

```diff
@@ -435,6 +459,26 @@ func (r *JobRunner) createEnvironment(ctx context.Context) ([]string, error) {
 		env["BUILDKITE_IGNORED_ENV"] = strings.Join(ignoredEnv, ",")
 	}
 
+	// Check if allowed environment variables are set
+	if len(r.conf.AgentConfiguration.AllowedEnvironmentVariables) > 0 {
+		// Fetch environment variables that are allowed by default
+		var allowedEnv = make([]string, len(DefaultAllowedEnv))
+		copy(allowedEnv, DefaultAllowedEnv)
+
+		var deniedEnv []string
+
+		// And add allowed environment variables from agent configuration
+		allowedEnv = append(allowedEnv, r.conf.AgentConfiguration.AllowedEnvironmentVariables...)
+
+		// Filter to get denied envs
+		deniedEnv = filterEnv(env, allowedEnv)
```

Comment:
remove
`var deniedEnv []string`
and make in-line:
`deniedEnv := filterEnv(env, allowedEnv)`

I would merge this together again:
```

// Set BUILDKITE_DENIED_ENV so the bootstrap can show warnings
deniedEnv := filterEnv(env, allowedEnv)
if len(deniedEnv) > 0 {
   	env["BUILDKITE_DENIED_ENV"] = strings.Join(deniedEnv, ",")
}
```

Reply:
Or even:
if deniedEnv := filterEnv(env, allowedEnv); len(deniedEnv) > 0 {
   	env["BUILDKITE_DENIED_ENV"] = strings.Join(deniedEnv, ",")
}